File: packages/sitecompiler-py/sitecompiler/resources/exports.py
import httpx
from ..exceptions import SiteCompilerError, PaymentRequiredError

class ExportsResource:
    def __init__(self, client: httpx.Client, base_url: str):
        self._client = client
        self._base_url = base_url

    def download_zip(self, job_id: str) -> bytes:
        res = self._client.get(f"{self._base_url}/api/job/{job_id}/download")
        if res.status_code == 403:
            raise PaymentRequiredError("Export pending payment approval", 403)
        if not res.is_success:
            raise SiteCompilerError(f"Failed to download zip: {res.text}", res.status_code)
        return res.content

    def get_site_model(self, job_id: str) -> dict:
        res = self._client.get(f"{self._base_url}/api/job/{job_id}/model")
        if not res.is_success:
            raise SiteCompilerError(f"Failed to fetch model: {res.text}", res.status_code)
        return res.json()

File: packages/sitecompiler-sdk/src/resources/exports.ts
import { PaymentRequiredError, SiteCompilerError } from '../errors';

export class ExportsResource {
  constructor(
    private readonly baseUrl: string,
    private readonly fetchFn: typeof fetch,
    private readonly headers: Record<string, string>
  ) {}

  /**
   * Download the compiled ZIP archive as an ArrayBuffer or Buffer
   */
  async downloadZip(jobId: string): Promise<ArrayBuffer> {
    const res = await this.fetchFn(`${this.baseUrl}/api/job/${jobId}/download`, {
      headers: this.headers,
    });

    if (res.status === 403) {
      throw new PaymentRequiredError();
    }

    if (!res.ok) {
      const errorData = await res.json().catch(() => ({}));
      throw new SiteCompilerError(errorData.error || 'Failed to download zip export', res.status, errorData);
    }

    return res.arrayBuffer();
  }

  /**
   * Retrieve the editable AST Site Model for a completed job
   */
  async getSiteModel(jobId: string): Promise<Record<string, unknown>> {
    const res = await this.fetchFn(`${this.baseUrl}/api/job/${jobId}/model`, {
      headers: this.headers,
    });

    if (!res.ok) {
      const errorData = await res.json().catch(() => ({}));
      throw new SiteCompilerError(errorData.error || 'Failed to fetch site model', res.status, errorData);
    }

    return res.json();
  }
}
